internal/server/router.go

```diff
@@ -16,6 +17,16 @@ import (
 func (s *Server) router() http.Handler {
 	r := chi.NewRouter()
 
+	if s.CustomNotFound != "" {
+		r.NotFound(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusNotFound)
+			if _, err := os.Stat(s.CustomNotFound); err == nil {
+				s.serveFile(s.CustomNotFound, w, r)
+				return
+			}
+		}))
+	}
+
```

Comment:
This is great and while I see what you did here, the server IMHO shouldn't start without prevalidating the file from existing.

This can be done [in this function](https://github.com/patrickdappollonio/http-server/blob/97d8e645e008e17edb63c448337ae3a8ab2dac3b/internal/server/validation.go#L15) where you can tap into `s.CustomNotFound` and check if the file exists in the filesystem. If not, then return an error.

For the rest of the validations in this part, while they are nice, they're also already inside the `s.serveFile()` function, so you don't need them here: https://github.com/patrickdappollonio/http-server/blob/97d8e645e008e17edb63c448337ae3a8ab2dac3b/internal/server/handlers.go#L178-L182

You can simply call `s.serveFile()` and call it a day.